core/new-gui/src/app/workspace/component/result-panel/result-panel.component.ts

```diff
@@ -1,15 +1,169 @@
+
+/**
+ * ResultPanelCompoent is the bottom level area that
+ *  displays the execution result of a workflow after
+ *  the execution finishes.
+ *
+ * The Component will display the result in an excel
+ *  table format, where each row represents a result
+ *  from the workflow and each column represents the
+ *  type of result the workflow returns.
+ *
+ * Clicking each row of the result table will create an
+ *  pop-up window and display the detail of that row
+ *  in a pretty json format.
+ *
+ * @author Henry Chen
+ * @author Zuozhi Wang
+ *
+ */
 @Component({
   selector: 'texera-result-panel',
   templateUrl: './result-panel.component.html',
   styleUrls: ['./result-panel.component.scss']
 })
 export class ResultPanelComponent implements OnInit {
 
-  constructor() { }
+  public showMessage: boolean = false;
+  public message: string = '';
+
+  public currentColumns: TableColumn[] | undefined;
+  public currentDisplayColumns: string[] | undefined;
+  public currentDataSource: MatTableDataSource<object> | undefined;
+  public currentDisplayRow: string = '';
+
+  @ViewChild(MatPaginator) paginator: MatPaginator | null = null;
+
+  constructor(private executeWorkflowService: ExecuteWorkflowService, private modalService: NgbModal) {
+    // once an execution has ended, update the result panel to dispaly
+    //  execution result or error
+    this.executeWorkflowService.getExecuteEndedStream().subscribe(
+      executionResult => this.handleResultData(executionResult),
+    );
+  }
+
+  /**
+   * Opens the ng-bootstrap model to display the row details in
+   *  pretty json format when clicked. User can view the details
+   *  in a larger, expanded format.
+   *
+   * @param content
+   */
+  public open(content: any): void {
+    this.modalService.open(content).result.then((result) => {
+      console.log(result);
+    }, (reason) => {
+      console.log(reason);
+    });
+  }
+
+
+  /**
+   * Fetches the information from the selected row in the result panel,
+   *  updates the content of the row to dispaly, and opens the
+   *  result panel to display the details of the selected rows.
+   *
+   * @param row
+   * @param content
+   */
+  public getRowDetails(row: any, content: any): void {
+    // console.log('getRowDetails: ');
+    // console.log(row);
+    this.currentDisplayRow = JSON.stringify(row, undefined, 2);
+    this.open(content);
+  }
 
   ngOnInit() {
+
   }
 
+  /**
+   *
+   * Update all the result table properties based on the newly acquired
+   *  execution result and display a new data table with a new paginator
+   *  on the result panel.
+   *
+   */
+  private changeResultTableProperty(response: SuccessExecutionResult) {
+    const resultData: ReadonlyArray<object> | undefined = response.result;
+
+    if (resultData !== undefined) {
+      // generate columnDef from first row, column definition is in order
+      this.currentDisplayColumns = Object.keys(resultData[0]).filter(x => x !== '_id');
```

Comment:
Please provide a brief overview in comments about what is being done here. An example of what comes as response and the transformation done would be great.

Reply:
Some additional comments have been added in the new commits.